e2e/etcd_test.go

```diff
@@ -176,7 +184,32 @@ func newEtcdProcessCluster(cfg *etcdProcessClusterConfig) (*etcdProcessCluster,
 			_, err := etcdp.proc.ExpectRegex(readyStr)
 			readyC <- err
 			etcdp.proc.ReadUntil('\n') // don't display rest of line
-			etcdp.proc.Interact()
+
+			done := make(chan struct{})
+			go func() {
+				buf := new(bytes.Buffer)
+				io.Copy(os.Stdout, buf)
+
+				var wg sync.WaitGroup
+				wg.Add(2)
+				go func() {
+					// grab stdin if any
+					io.Copy(buf, os.Stdin)
+					wg.Done()
+				}()
+				go func() {
+					// stdin to buffer
+					// buffer to stdout
+					io.Copy(os.Stdout, buf)
+					wg.Done()
+				}()
+				wg.Wait()
+
+				etcdp.proc.Cmd.Wait()
+				done <- struct{}{}
+			}()
+			<-done
```

Comment:
This is replacing https://github.com/coreos/gexpect/blob/master/gexpect.go#L373-L378:

```go
func (expect *ExpectSubprocess) Interact() {
	defer expect.Cmd.Wait()
	io.Copy(os.Stdout, &expect.buf.b)
	go io.Copy(os.Stdout, expect.buf.f)
	go io.Copy(expect.buf.f, os.Stdin)
}
```